Foundation/NSData.swift

```diff
@@ -435,6 +435,24 @@ open class NSData : NSObject, NSCopying, NSMutableCopying, NSSecureCoding {
 
     internal func makeTemporaryFile(inDirectory dirPath: String) throws -> (Int32, String) {
         let template = dirPath._nsObject.appendingPathComponent("tmp.XXXXXX")
+#if os(Windows)
```

Comment:
Couldnt this whole function just be replaced with:
`return try _NSCreateTemporaryFile(dirPath._nsObject.appendingPathComponent("")`
or something like that, they seem to have the same bodies of code expect for computing the initial template.

Reply:
I honestly don't know.  I'm not sure why this is duplicated in the first place, as the same thing happens in the Linux case.

Reply:
What about passing the `template` as the argument, eg:

`internal func _NSCreateTemporaryFile(withTemplate template: String)`

There only seems to be one caller of each method so generating the template at the call-site seems to solve the issue.